bgpd/bgp_route.c

```diff
@@ -1435,9 +1435,10 @@ subgroup_announce_check (struct bgp_info *ri, struct update_subgroup *subgrp,
 
 #ifdef HAVE_IPV6
 #define NEXTHOP_IS_V6 (\
-    (safi != SAFI_ENCAP && \
+    (safi != SAFI_ENCAP && safi != SAFI_MPLS_VPN &&\
      (p->family == AF_INET6 || peer_cap_enhe(peer))) || \
-    (safi == SAFI_ENCAP && attr->extra->mp_nexthop_len == 16))
+    ((safi == SAFI_ENCAP || safi != SAFI_MPLS_VPN) &&\
+     attr->extra->mp_nexthop_len == 16))
 
```

Comment:
in bgp_route.h
#define BGP_NEXTHOP_AFI_FROM_NHLEN(nhlen) \
  ((nhlen) < IPV4_MAX_BYTELEN  ? 0 : \
   ((nhlen) < IPV6_MAX_BYTELEN ? AFI_IP : AFI_IP6))